File: example_00/bubble_sort.py
import time
import sys

DIR = 'archivos_txt/'

def main():
    txt= sys.argv[1] 
    #abrimos archivo
    archivo = open(DIR+txt, "r")
    #parseamos por linea para convertir en un array
    numeros = archivo.read().splitlines()
    #cerramos archivo
    archivo.close()
    #print(numeros)

    inicio = time.time()
    bubbleSort(numeros)
    fin = time.time()
    print(str(fin-inicio))
    #print(numeros)

def bubbleSort(numeros):
    n = len(numeros)

    for i in range(n):

        for j in range(0, n-i-1):
            if numeros[j] > numeros[j+1] :
                numeros[j], numeros[j+1] = numeros[j+1], numeros[j]

if __name__ == '__main__':
   main()

File: example_00/bubble_sort.cpp
#include <iostream>
#include <string>
#include <fstream>
#include <bits/stdc++.h>

#define     PATH "archivos_txt/"

using std::cout;
using std::cin;
using std::endl;


clock_t inicio, fin;
double tiempo_ejecucion;

void swap(int *xp, int *yp)
{
    int temp = *xp;
    *xp = *yp;
    *yp = temp;
}
 
// A function to implement bubble sort
void bubbleSort(int arr[], int n)
{
    

    inicio = clock();
    int i, j;
    for (i = 0; i < n-1; i++)    
        for (j = 0; j < n-i-1; j++)
            if (arr[j] > arr[j+1])
                swap(&arr[j], &arr[j+1]);
    fin = clock();
    
    tiempo_ejecucion = double(fin - inicio) / double(CLOCKS_PER_SEC);
}
 

void printArray(int arr[], int size)
{
    int i;
    for (i = 0; i < size; i++)
        cout << arr[i] << " ";
    cout << endl;
}

std::string toString(char* txt){
    std::stringstream sti;
    std::string SIZE;
        sti << txt;
        sti >> SIZE;
        return SIZE+".txt";
}

int main(int nargs, char *argv[])
{
    const int tam = atoi(argv[1]);;
    int arreglo[tam];
    int element;
    std::ifstream in;
    std::string cantidad_txt, nombre_archivo;
    cantidad_txt = toString(argv[1]);
    nombre_archivo = PATH+cantidad_txt;
    in.open(nombre_archivo);

    if (in.is_open()) {
        int i = 0;
        while (in >> element) {
            arreglo[i] = element;
            i++;
        }
    }
    in.close();
    bubbleSort(arreglo, tam);
    cout<<tiempo_ejecucion<< endl;
    return 0;
}